### DISCUSSION TITLE
How to allocate perpetual tabs like i3 numbered workspace?

### DISCUSSION THREAD

**Discussion by otakutyrant** [kovidgoyal/kitty#6405] [Q&A] [answered]
In i3, ten workspaces are allocated to begin with and reachable by number 1~10. User just jump to any specific workspace with `$mod+{1~10}`, and they do not need to create new workspace manually in advance. When there is not any window in a numbered workspace, this workspace hide from the status bar.

Now I want to accomplish the same effect in kitty. I wrote a session:

```
layout horizontal
cd ~
launch zsh
launch zsh
new_tab
layout horizontal
launch
launch
new_tab Projects
layout horizontal
cd ~/Projects
launch
new_tab
new_tab
new_tab
new_tab
new_tab
new_tab
new_tab
```

However there are still some issues to solve:

1. `new_tab` will launch a shell anyway. Suppossing Kitty allows hiding a tab, Kitty still do not know whether should to hide the tab because the later always has a shell.
2. When exit the shell of a numbered tab, this tab will vanish and be replaced by the next tab. I do not know how to make it perpetual.

If Kitty has no related feature, I can try to make PR.

**Comment by kovidgoyal** [comment, marked as answer]
There is no such thing as a perpetual tab. Any window in kitty has to be
associated with a running a program. If you quit the program the window
will close or be unuseable, since no program is asosciated witht he tty
device corresponding to the window.

And I dont understand what your use case is. Why do you want pre-created
tabs? Simply create a tab whenever you need one.

**Reply by otakutyrant** [reply]
What you said makes sense. I decide to allocate only three tabs in my startup session.
